rbi/core/comparable.rbi

```diff
@@ -52,12 +52,12 @@ module Comparable
 
   # Compares two objects based on the receiver's `<=>` method, returning true if
   # it returns 1.
-  sig { params(other: T.untyped).returns(T::Boolean) }
+  sig { params(other: T.self_type).returns(T::Boolean) }
   def >(other); end
 
   # Compares two objects based on the receiver's `<=>` method, returning true if
   # it returns 0 or 1.
-  sig { params(other: T.untyped).returns(T::Boolean) }
+  sig { params(other: T.self_type).returns(T::Boolean) }
```

Comment:
I think these should probably be comparing against `BasicObject` instead of `T.self_type` as there are places in the stdlib where comparison is allowed to other types.

Reply:
+1

Reply:
It turns out that at Stripe we even rely on this behavior (our "monetary amount" type allows comparisons against the literal `0`). I'd rather not have to change Stripe's codebase for this, given that for things like `==` and `<=>` on `Kernel` the RBI already has `BasicObject` for `arg0`.